pkg/routes/middleware.go

```diff
@@ -44,3 +44,36 @@ func ESMiddleware(c *gin.Context) {
 	// If not, it will not work, and show "The client noticed that the server is not Elasticsearch and we do not support this unknown product."
 	c.Header("X-elastic-product", "Elasticsearch")
 }
+
+func IndexAliasMiddleware(c *gin.Context) {
+	target := ""
+	ix := 0
+
+	for i, entry := range c.Params {
+		if entry.Key == "target" {
+			target = entry.Value
+			ix = i
+			break
+		}
+	}
+
+	if target == "" {
+		c.Next()
+		return
+	}
+
+	indexList := core.ZINC_INDEX_LIST.List()
+	newTarget := ""
+
+	// find all index that match this alias and add them to the newTarget
+	for _, index := range indexList {
```

Comment:
This will be a problem if we have 60,000 indexes. One user actually has 60,000 indexes.